src/map/skill.c

```diff
@@ -13362,30 +13362,34 @@ static int skill_unit_onplace_timer(struct skill_unit *src, struct block_list *b
 
 		case UNT_ANKLESNARE:
 		case UNT_MANHOLE:
-			if( sg->val2 == 0 && tsc && (sg->unit_id == UNT_ANKLESNARE || bl->id != sg->src_id) ) {
-				int sec = skill->get_time2(sg->skill_id,sg->skill_lv);
-				if( status->change_start(ss,bl,type,10000,sg->skill_lv,sg->group_id,0,0,sec, SCFLAG_FIXEDRATE) ) {
-					const struct TimerData* td = tsc->data[type]?timer->get(tsc->data[type]->timer):NULL;
-					if( td )
+			if (sg->val2 == 0 && tsc && (sg->unit_id == UNT_ANKLESNARE || bl->id != sg->src_id)) {
+				int sec = skill->get_time2(sg->skill_id, sg->skill_lv);
+				if (bl->type == BL_MOB && map->list[bl->m].flag.gvg_castle == 0) {
+					struct mob_data *md = BL_UCAST(BL_MOB, bl);
+					if (sg->unit_id == UNT_MANHOLE && (md->class_ == MOBID_EMPELIUM || md->class_ == MOBID_BARRICADE || md->class_ == MOBID_S_EMPEL_1 || md->class_ == MOBID_S_EMPEL_2)) {
+						// Do nothing...
+					}
```

Comment:
Hi MrKeiKun

I think this code is causing #2964

can you explain what this if block is doing ?  Because as soon as

```
if (bl->type == BL_MOB && map->list[bl->m].flag.gvg_castle == 0) {
```

resolves to true, the rest seem to matter.  And this resolves to true when anklesnare hits a mob on any non-gvg_castle map